packages/core/src/routes/experience/classes/experience-interaction.ts

```diff
@@ -187,8 +189,18 @@ export default class ExperienceInteraction {
     this.verificationRecords.set(type, record);
   }
 
-  public getVerificationRecordById(verificationId: string) {
-    return this.verificationRecordsArray.find((record) => record.id === verificationId);
+  public getVerificationRecordByTypeAndId<K extends keyof VerificationRecordMap>(
+    type: K,
+    verificationId: string
+  ) {
+    const record = this.verificationRecords.get(type);
+
+    assertThat(
+      record?.id === verificationId,
+      new RequestError({ code: 'session.verification_session_not_found', status: 404 })
```

Comment:
this looks ambiguous. maybe a new code?

Reply:
This is aligned with the current interaction API error code. An issue is created tracking this. Revisit the error codes later.